### ISSUE TITLE
Built-in focus styles in Editable

### ISSUE THREAD

**Issue by Chudesnov** [ianstormtaylor/slate#5146]
### Description

`<Editable>` has the role of [textbox](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/textbox_role) and is the focus target for keyboard users, so it should have some styles indicating when it's focused. Browser provides a built-in focus outline when `contenteditable` attribute is set on an element https://github.com/ianstormtaylor/slate/blob/0eb37e79150275d3535f1694d8972751a83d826f/packages/slate-react/src/components/editable.tsx#L842-L843 but due to this line https://github.com/ianstormtaylor/slate/blob/0eb37e79150275d3535f1694d8972751a83d826f/packages/slate-react/src/components/editable.tsx#L853-L854 the built-in style is removed.

### Workaround
```<Editable style={{outline: undefined}} />``` resets the value set in `editable.tsx` and reverts to default, user or author styles set through CSS.

### Proposed solution
There's no good reason to "prevent the default outline styles" by default. Remove these lines https://github.com/ianstormtaylor/slate/blob/0eb37e79150275d3535f1694d8972751a83d826f/packages/slate-react/src/components/editable.tsx#L853-L854

If a particular styling technique (displaying custom focus styles around the whole editor area including its toolbar) is preferred, it should be described in the relevant documentation article rather than being partially hard-coded in the `Editable` component.
